distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/FederatedCswServer.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.restito;
+
+import static org.junit.Assert.fail;
+import static com.xebialabs.restito.semantics.Action.bytesContent;
+import static com.xebialabs.restito.semantics.Action.contentType;
+import static com.xebialabs.restito.semantics.Action.ok;
+import static com.xebialabs.restito.semantics.Condition.get;
+import static com.xebialabs.restito.semantics.Condition.post;
+import static com.xebialabs.restito.semantics.Condition.withPostBodyContaining;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import com.xebialabs.restito.builder.stub.StubHttp;
+import com.xebialabs.restito.builder.verify.VerifyHttp;
+import com.xebialabs.restito.server.StubServer;
+
+/**
+ * Encompasses a Restito CSW Stub Server for use in integration testing. Handles handshake responses
+ * and federated query responses for a csw endpoint.
+ */
+public class FederatedCswServer {
+    private static final XLogger LOGGER =
+            new XLogger(LoggerFactory.getLogger(FederatedCswServer.class));
+
+    private final String sourceId;
+
+    private final String httpRoot;
+
+    private final int port;
+
+    private StubServer cswStubServer;
```

Comment:
The methods we have should be enough for testing, but I went ahead and added `getServer` since there are a few other public methods on StubServer has that may be needed for some reason.